src/pudl/convert/datapkg_to_rst.py

```diff
@@ -0,0 +1,130 @@
+"""
+Module to convert json metadata into rst files.
+
+All of the information about the transformed pudl tables, namely their fields
+types and descriptions, resides in the datapackage metadata. This module makes
+that information available to users, without duplicating any data, by converting
+json metadata files into documentation-compatible rst files. The functions
+serve to extract the field names, field data types, and field descriptions of
+each pudl table and outputs them in a manner that automatically updates the
+read-the-docs.
+"""
+
+import argparse
+import json
+import logging
+import sys
+
+from jinja2 import Template
+
+logger = logging.getLogger(__name__)
+
+
+###############################################################################
+# T E M P L A T E S
+###############################################################################
+
+"""
+The following templates map json data into one long rst file seperated by table
+titles and document links (RST_TEMPLATE)
+
+It's important for the templates that the json data do not contain excess
+white space either at the beginning or the end of each value.
+"""
+
+
+# Template for all tables in one rst file
+RST_TEMPLATE = '''
+===============================================================================
+All PUDL Database Tables
+===============================================================================
+
+{% for resource in resources %}
+.. _{{ resource.name }}:
+
+-------------------------------------------------------------------------------
+{{ resource.name }}
+-------------------------------------------------------------------------------
```

Comment:
I created a top level header that's different from the many individual table headers so that they can nest under it in the navigation sidebar. I also changed the title of the table sections to just be the table name, since they were all running to more than a single line in the sidebar, making it cluttered and repetitive.